stable/jenkins/README.md

```diff
@@ -516,6 +516,59 @@ agent:
 ```
 Best reference is https://<jenkins_url>/configuration-as-code/reference#Cloud-kubernetes.
 
+### Adding pod templates using additional `agent` values (multiple agents)
+
+Additional `agent` values may be used to configure kubernetes pod templates in the default configuration as code. For example,
+```yaml
+master:
+  JCasC:
+    enabled: true
+    defaultConfig: true
+
+agent:
+  enabled: true
+  podName: default
+  customJenkinsLabels: default
+  resources:
+    limits:
+      cpu: "1"
+      memory: "2048Mi"
+  podTemplates:
+    additionalAgents: |-
+      {{- /* save .Values.agent */}}
+      {{- $agent := .Values.agent }}
+
+      {{- range $name, $additionalAgent := .Values.additionalAgents }}
+        {{- /* merge original .Values.agent into additional agent to ensure it at least has the default values */}}
+        {{- $additionalAgent := merge $additionalAgent $agent }}
+        {{- /* set .Values.agent to $additionalAgent */}}
+        {{- $_ := set $.Values "agent" $additionalAgent }}
+        {{- include "jenkins.casc.podTemplate" $ | nindent 0 }}
+      {{- end }}
+
+      {{- /* restore .Values.agent */}}
+      {{- $_ := set .Values "agent" $agent }}
```

Comment:
FYI. In an attempt to provide a layer of abstraction, I moved this whole templating block to a helper template and changed the above to simply
`{{- include "jenkins.casc.podTemplates" .Values.additionalAgents }}`

Not surprisingly I encountered scoping issues. See https://github.com/helm/helm/issues/3167
```
{{/*
Returns a list of kubernetes pod templates configured from a list of agent values
*/}}
{{- define "jenkins.casc.podTemplates" -}}
    {{- /* save .Values.agent */}}
    {{- $agent := $.Values.agent }}

    {{- range $name, $additionalAgent := . }}
      {{- /* set .Values.agent to $additionalAgent */}}
      {{- $_ := set $.Values "agent" $additionalAgent }}
      {{- include "jenkins.casc.podTemplate" $ | nindent 0 }}
    {{- end }}

    {{- /* restore .Values.agent */}}
    {{- $_ := set $.Values "agent" $agent }}
{{- end -}}
```
> template: jenkins/templates/jcasc-config.yaml:6:4: executing "jenkins/templates/jcasc-config.yaml" at <include "jenkins.casc.podTemplates" .Values.additionalAgents>: error calling include: template: jenkins/templates/_helpers.tpl:182:19: executing "jenkins.casc.podTemplates" at <$.Values.agent>: nil pointer evaluating interface {}.agent